### PR TITLE
[Feature] Add reverse edges in CitationGraphDataset

### PR DESCRIPTION
**Pull request by JuliaSun623**
## Description
This pr adds optional parameter "reverse_edge (bool)" in CitetationGraphDataset, determining whether to reverse edges in graph. Related issue: https://github.com/dmlc/dgl/issues/2580

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by HuXiangkun — CHANGES_REQUESTED** [review]

**Review by JuliaSun623 — COMMENTED** [review]
reviewed

File: python/dgl/data/dgl_dataset.py [hunk 1]
```diff
@@ -49,6 +49,8 @@ class DGLDataset(object):
         Whether to reload the dataset. Default: False
     verbose : bool
         Whether to print out progress information
+    reverse_edge: bool
```
Comment (hetong007) [inline python/dgl/data/dgl_dataset.py:52]: This option is only valid for citation graphs, we won't add it to the base class.

**Review by hetong007 — CHANGES_REQUESTED** [review]

File: python/dgl/data/citation_graph.py [hunk 2]
```diff
@@ -104,7 +107,11 @@ def process(self):
 
         features = sp.vstack((allx, tx)).tolil()
         features[test_idx_reorder, :] = features[test_idx_range, :]
-        graph = nx.DiGraph(nx.from_dict_of_lists(graph))
+
+        if self.reverse_edge == True:
```
Comment (VoVAllen) [inline python/dgl/data/citation_graph.py:111]:
```suggestion
        if self.reverse_edge:
```

**Review by hetong007 — APPROVED** [review]
LGTM

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**